api/testutils/kusttest/harnessenhanced.go

```diff
@@ -109,12 +114,69 @@ func (th *HarnessEnhanced) LoadAndRunTransformer(
 	return resMap
 }
 
+func (th *HarnessEnhanced) TestTransformer(
+	config, input, expected string) {
+	for _, b := range []bool{true, false} {
+		th.t.Run(fmt.Sprintf("yaml-%v", b), func(t *testing.T) {
+			c, err := amendConfig(config, b)
+			if err != nil {
+				th.t.Fatalf("Err: %v", err)
+			}
+			resMap, err := th.RunTransformer(c, input)
+			if err != nil {
+				th.t.Fatalf("Err: %v", err)
+			}
+			th.AssertActualEqualsExpected(resMap, expected)
+		})
+	}
+}
+
+func amendConfig(config string, yamlSupport bool) (string, error) {
```

Comment:
`toggleYamlSupportFlag`

(So much code to set a boolean :)

Reply:
or `Field` instead of flag.

Reply:
done